paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -254,17 +265,29 @@ public void commit(ManifestCommittable committable, Map<String, String> properti
                 // assume this compact commit follows just after the append commit created above
                 safeLatestSnapshotId += 1;
             }
-
-            tryCommit(
-                    compactTableFiles,
-                    compactChangelog,
-                    Collections.emptyList(),
-                    committable.identifier(),
-                    committable.watermark(),
-                    committable.logOffsets(),
-                    Snapshot.CommitKind.COMPACT,
-                    safeLatestSnapshotId);
-        }
+            generatedSnapshot += 1;
+            attempts +=
+                    tryCommit(
+                            compactTableFiles,
+                            compactChangelog,
+                            Collections.emptyList(),
+                            committable.identifier(),
+                            committable.watermark(),
+                            committable.logOffsets(),
+                            Snapshot.CommitKind.COMPACT,
+                            safeLatestSnapshotId);
+        }
+        long commitDuration = (System.nanoTime() - started) / 1_000_000;
+        CommitStats commitStats =
+                new CommitStats(
+                        appendTableFiles,
+                        appendChangelog,
+                        compactTableFiles,
+                        compactChangelog,
+                        commitDuration,
+                        generatedSnapshot,
+                        attempts);
+        commitMetrics.reportCommit(commitStats);
```

Comment:
I'm confused here, if one of the snapshot failed, then it could be regarded as a complete commit? And if the `APPEND` snapshot failed due to conflicts, should the statistics been reported?

Reply:
If `APPEND` succeeds and `COMPACT` fails it is still a successful commit. If `APPEND` fails I think it should still count as an attempt, though not a successful attempt.